okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -224,13 +227,23 @@ open class OkHttpClient internal constructor(
   constructor() : this(Builder())
 
   init {
+    if (builder.trustManagerOverrides != null) {
+      builder.x509TrustManagerOrNull = TrustManagerBridge.Builder()
+          .addOverrides(builder.trustManagerOverrides!!)
+          .apply {
+            if (builder.x509TrustManagerOrNull != null) {
+              this.default(builder.x509TrustManagerOrNull!!)
+            }
+          }
+          .build()
+    }
+
     if (builder.sslSocketFactoryOrNull != null || connectionSpecs.none { it.isTls }) {
       this.sslSocketFactoryOrNull = builder.sslSocketFactoryOrNull
       this.certificateChainCleaner = builder.certificateChainCleaner
       this.x509TrustManager = builder.x509TrustManagerOrNull
     } else {
-      this.x509TrustManager = Platform.get().platformTrustManager()
-      Platform.get().configureTrustManager(x509TrustManager)
+      this.x509TrustManager = builder.x509TrustManagerOrNull ?: Platform.get().platformTrustManager()
```

Comment:
Provided we are ok with this line, the whole change *should* be a noop if the new method isn't used.  Consider that in the scope of the change.